### PR TITLE
WIP detecting spikes in graphite metrics

### PR DESCRIPTION
**Pull request by wheresrhys**

### CODE REVIEW

File: src/checks/aggregate.check.js [hunk 1]
```diff
@@ -36,11 +34,8 @@ class AggregateCheck extends Check {
 	start(){
 		let watchRegex = new RegExp(`(${this.watch.join('|')})`, 'i');
 		this.obserables = this.parent.checks.filter(check => watchRegex.test(check.name));
-		this.int = setInterval(this.tick.bind(this), ms(this.interval || '60s'));
```
Comment (aintgoin2goa) [inline src/checks/aggregate.check.js]: why these changes?  this is gonna break stuff

File: src/checks/graphiteSpike.check.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+				});
+			})
+			.then(data => {
+				let ok;
+				if (this.direction === 'up') {
+					ok = data.sample / data.baseline < this.threshold;
+				} else {
+					ok = data.sample / data.baseline > 1 / this.threshold;
+				}
+				this.status = ok ? status.PASSED : status.FAILED;
```
Comment (aintgoin2goa) [inline src/checks/graphiteSpike.check.js]: You also need to add a checkOutput option which explains what has happened / why it failed

Comment (wheresrhys) [inline src/checks/graphiteSpike.check.js]: oh, didn't know that was compulsory. Will try to write a test for that then add to graphiteSpike

Comment (aintgoin2goa) [inline src/checks/graphiteSpike.check.js]: I think that the base class might provide you with some sort of default, but I'm not sure.  Is worth having something explicit anyway
